core/views.py

```diff
@@ -112,3 +122,39 @@ class HackathonsRUDView(generics.RetrieveUpdateDestroyAPIView):
     serializer_class = HackathonSerializer
     lookup_field = 'pk'
     queryset = Hackathon.objects.all()
+
+
+class HackathonSubmissionView(generics.ListCreateAPIView):
+    """
+    API used to get the list of all the submissions of particular hackathon.
+    """
+    serializer_class = SubmissionsSerializer
+
+    def get_permissions(self):
+        return [permissions.AllowAny()]
+        # try:
+        #     hackathon = Hackathon.objects.get(id=self.kwargs['pk'])
+        #     if self.request.method == 'GET' and hackathon.status!="Completed":
+        #         return [permissions.IsAdminUser()]
+        #     return [permissions.AllowAny()]
+        # except Hackathon.DoesNotExist:
+        #     raise exceptions.NotFound("Hackathon does not exist!")
+
```

Comment:
There's no need for this function. By default, this view supports the same permissions.

Reply:
Actually before I was giving some permissions so instead of removing the function I commented the unnecessary part.

Reply:
Okay, np.